File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/alarm/repository/AlarmRepository.java
package cc.nobrain.dev.userserver.domain.alarm.repository;

import cc.nobrain.dev.userserver.domain.alarm.entity.Alarm;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;

public interface AlarmRepository extends JpaRepository<Alarm, Long>, JpaSpecificationExecutor, QuerydslPredicateExecutor<Alarm>, AlarmRepositoryCustom {

}

File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/alarm/repository/AlarmRepository.kt
package cc.nobrain.dev.userserver.domain.alarm.repository

import cc.nobrain.dev.userserver.domain.alarm.entity.Alarm
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.querydsl.QuerydslPredicateExecutor

interface AlarmRepository : JpaRepository<Alarm, Long>, JpaSpecificationExecutor<Alarm>, QuerydslPredicateExecutor<Alarm>, AlarmRepositoryCustom